Yj/cla test

yjjiang11 opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
Others
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
Others
### Describe
<!-- Describe what this PR does -->
test

Commit:
rm eager guard tests for imperative ops

Commit:
lint

Commit:
minor change

yjjiang11 committed:
Merge branch 'develop' of github.com:PaddlePaddle/Paddle into yj/rm_imperative_eager_guard

yjjiang11 committed:
Merge branch 'develop' into yj/rm_imperative_eager_guard

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=48759) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=48759) before we can accept your contribution.<br/><hr/>**yjjiang11** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle?pullRequest=48759) it.</sub>

yjjiang11 committed:
add patch

yjjiang11 committed:
rm

yjjiang11 committed:
modified:   cmake/external/dgc.cmake
	modified:   paddle/phi/kernels/onednn/matmul_grad_kernel.cc
	modified:   python/paddle/fluid/contrib/layers/rnn_impl.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_imperative_out_scale.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_imperative_ptq.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_imperative_qat.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_imperative_qat_amp.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_imperative_qat_lsq.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_imperative_qat_user_defined.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_imperative_skip_op.py
	modified:   python/paddle/fluid/contrib/slim/tests/test_quantization_pass.py
	modified:   python/paddle/fluid/dygraph/rnn.py
	modified:   python/paddle/fluid/layers/detection.py
	modified:   python/paddle/fluid/layers/nn.py
	modified:   python/paddle/fluid/nets.py
	modified:   python/paddle/fluid/tests/test_detection.py
	modified:   python/paddle/fluid/tests/unittests/auto_parallel/test_dist_matmul.py
	modified:   python/paddle/fluid/tests/unittests/auto_parallel/test_dist_op_cost.py
	modified:   python/paddle/fluid/tests/unittests/auto_parallel_gpt_model.py
	modified:   python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_pp_embedding.py
	modified:   python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_pp_recompute.py
	modified:   python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_pp_transformer.py
	modified:   python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_pp_transformer_with_virtual_stage.py
	modified:   python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_shared_weight.py
	modified:   python/paddle/fluid/tests/unittests/collective/fleet/parallel_dygraph_transformer.py
	modified:   python/paddle/fluid/tests/unittests/dist_transformer.py
	modified:   python/paddle/fluid/tests/unittests/dygraph_to_static/bert_dygraph_model.py
	modified:   python/paddle/fluid/tests/unittests/dygraph_to_static/seq2seq_dygraph_model.py
	modified:   python/paddle/fluid/tests/unittests/dygraph_to_static/test_bmn.py
	modified:   python/paddle/fluid/tests/unittests/dygraph_to_static/test_dict.py
	modified:   python/paddle/fluid/tests/unittests/dygraph_to_static/test_program_translator.py
	modified:   python/paddle/fluid/tests/unittests/dygraph_to_static/test_ptb_lm.py
	modified:   python/paddle/fluid/tests/unittests/dygraph_to_static/transformer_dygraph_model.py
	modified:   python/paddle/fluid/tests/unittests/ipu/test_matmul_op_ipu.py
	modified:   python/paddle/fluid/tests/unittests/ipu/test_weight_sharing_ipu.py
	modified:   python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_matmul_op_output_fuse_pass.py
	modified:   python/paddle/fluid/tests/unittests/ir/inference/test_trt_inspector.py
	modified:   python/paddle/fluid/tests/unittests/ir/inference/test_trt_matmul.py
	modified:   python/paddle/fluid/tests/unittests/ir/inference/test_trt_matmul_quant_dequant.py
	modified:   python/paddle/fluid/tests/unittests/mkldnn/check_flags_mkldnn_ops_on_off.py
	modified:   python/paddle/fluid/tests/unittests/mkldnn/test_flags_mkldnn_ops_on_off.py
	modified:   python/paddle/fluid/tests/unittests/parallel_dygraph_sparse_embedding.py
	modified:   python/paddle/fluid/tests/unittests/test_auto_parallel_completion.py
	modified:   python/paddle/fluid/tests/unittests/test_auto_parallel_completion_gpt.py
	modified:   python/paddle/fluid/tests/unittests/test_auto_parallel_partitioner.py
	modified:   python/paddle/fluid/tests/unittests/test_auto_parallel_partitioner_gpt.py
	modified:   python/paddle/fluid/tests/unittests/test_auto_search_dist_matmul_op.py
	modified:   python/paddle/fluid/tests/unittests/test_auto_search_dist_op.py
	modified:   python/paddle/fluid/tests/unittests/test_cholesky_op.py
	modified:   python/paddle/fluid/tests/unittests/test_dist_transpiler.py
	modified:   python/paddle/fluid/tests/unittests/test_eager_deletion_padding_rnn.py
	modified:   python/paddle/fluid/tests/unittests/test_fused_attention_op.py
	modified:   python/paddle/fluid/tests/unittests/test_fused_multi_transformer_int8_op.py
	modified:   python/paddle/fluid/tests/unittests/test_fused_multi_transformer_op.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_basic.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_gnn.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_lod_tensor_to_selected_rows.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_save_load.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_save_load_v2.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_selected_rows_to_lod_tensor.py
	modified:   python/paddle/fluid/tests/unittests/test_imperative_transformer_sorted_gradient.py
	modified:   python/paddle/fluid/tests/unittests/test_layers.py
	modified:   python/paddle/fluid/tests/unittests/test_matmul_op.py
	modified:   python/paddle/fluid/tests/unittests/test_mul_nn_grad.py
	modified:   python/paddle/fluid/tests/unittests/test_recurrent_op.py
	modified:   python/paddle/fluid/tests/unittests/test_rnn_decode_api.py
	modified:   python/paddle/fluid/tests/unittests/test_rpn_target_assign_op.py
	modified:   python/paddle/fluid/tests/unittests/test_static_save_load.py
	modified:   python/paddle/fluid/tests/unittests/transformer_model.py
	modified:   python/paddle/fluid/tests/unittests/xpu/test_fused_attention_op_xpu.py
	modified:   python/paddle/fluid/tests/unittests/xpu/test_matmul_op_xpu.py
